app/src/main/java/com/github/onlynotesswent/model/note/Note.kt

```diff
@@ -65,4 +67,96 @@ data class Note(
       val classYear: Int,
       val publicPath: String
   )
+
+  /**
+   * Represents a list of Comments for a Note. The class is immutable. It's companion object is used
+   * to produce a new CommentCollection to get a new one.
+   */
+  class CommentCollection(val commentsList: List<Comment> = emptyList()) {
+    fun getUserComments(userId: String): List<Comment> {
+      return commentsList.filter { it.userId == userId }
+    }
+
+    companion object {
+
+      /**
+       * Generates a unique ID based on the user ID, content, and current timestamp. (There is
+       * nearly no risk of collisions as a user would have to input the exact same comment content
+       * at the exact same millisecond under a Note for both comments to have the exact same Id)
+       *
+       * @param userId The unique identifier for the user.
+       * @param content The content of the comment.
+       * @return A SHA-256 hash string that uniquely represents this comment instance.
+       */
+      private fun generateId(userId: String, content: String): String {
+        val timestamp = System.currentTimeMillis()
+        val input = "$userId$content$timestamp"
+        val digest = MessageDigest.getInstance("SHA-256")
+        val hashBytes = digest.digest(input.toByteArray())
+        return hashBytes.joinToString("") { "%02x".format(it) }
+      }
+      /**
+       * Adds a new comment to the CommentCollection with a unique ID.
+       *
+       * @param userId The ID of the user adding the comment.
+       * @param content The text content of the comment.
+       * @param CommentCollection The current collection of comments.
+       * @return An updated CommentCollection including the new comment.
+       */
+      fun addComment(
+          userId: String,
+          content: String,
+          CommentCollection: CommentCollection
+      ): CommentCollection {
+        val mutableCommentsList = CommentCollection.commentsList.toMutableList()
+        mutableCommentsList.add(0, Comment(generateId(userId, content), userId, content))
+        return CommentCollection(mutableCommentsList.toList())
+      }
+      /**
+       * Edits an existing comment in the CommentCollection by its ID.
+       *
+       * @param commentId The ID of the comment to be edited.
+       * @param content The new content for the comment.
+       * @param CommentCollection The current collection of comments.
+       * @return An updated CommentCollection with the modified comment.
+       */
+      fun editComment(
+          commentId: String,
+          content: String,
+          CommentCollection: CommentCollection
+      ): CommentCollection {
+        val mutableCommentsList = CommentCollection.commentsList.toMutableList()
+        return CommentCollection(
+            mutableCommentsList.map {
+              if (it.commentId == commentId)
+                  Comment(generateId(it.userId, content), it.userId, content)
```

Comment:
As a suggestion it might be better to not modify the existing id of the comment and just reuse it. But I am not so sure which option might work better, so we can leave it like this and discuss it with the rest of the team to see which option to choose.